types/user.ts

```diff
@@ -4,24 +4,26 @@ import { DocumentReference, DocumentData } from 'firebase/firestore';
 import { Send } from './types';
 
 export class User extends LazyObject {
-  protected username: string | undefined;
-  protected bio: string | undefined;
-  protected status: UserStatus | undefined;
-  protected sends: Send[] | undefined;
-  protected following: User[] | undefined;
-  protected followers: User[] | undefined;
+  protected username?: string;
+  protected email?: string;
+  protected bio?: string;
+  protected status?: UserStatus;
+  protected sends?: Send[];
+  protected following?: User[];
+  protected followers?: User[];
 
   protected initWithDocumentData(data: DocumentData): void {
     this.username = data.username;
+    this.email = data.email;
     this.bio = data.bio;
     this.status = data.status as UserStatus;
-    this.sends = data.sends.map(
+    this.sends = (data.sends || []).map(
```

Comment:
Although theoretically valid, this syntax isn't the most correct for this scenario. Consider if `sends` were a boolean, then this statement would return the right hand side even though `sends` exists. Since what we really care about here is whether or not `sends` *exists*, we can take advantage of the null or default syntax: `??`
```suggestion
    this.sends = (data.sends ?? []).map(
```